### ISSUE TITLE
normalization

### ISSUE THREAD

**Issue by LaiSongxuan** [Theano/Theano#5167]
let x be the input vector. let std = tensor.std(x). Then std is a function of x. How can I make std preserve its value while make it not a function of x? It seems when applying batch normalization, std(x) will backprop gradients?

**Comment by nouiz** [comment]
for user questions, as on theano-user mailing list.

I don't understand what you mean by make std not a function of x. If you do:

x=theano.tensor.matrix()
o=theano.tensor.std(x)

then o will always depend of x. The only way to change that is to clone the graph while  changing it with the replace parameter or via the givens parameter of theano.function().

Try to clarify want you want in the email.
